### PR TITLE
fixes #1832: Adds ability to get WKT for lat/lon geo functions.

### LINKED ISSUE #1832: Update GeoFeatureVisitor to handle geo functions against separate lat/lon fields
If we encounter one of these functions, we currently throw an exception which is bad.  We should be able to create WKT to display on the map for these functions the same as we do for any other geo function.

### PR DESCRIPTION
**Pull request by jwomeara**

### CODE REVIEW

**Review by drewfarris — DISMISSED** [review]

File: warehouse/query-core/src/main/java/datawave/query/jexl/functions/GeoFunctionsDescriptor.java [hunk 1]
```diff
@@ -305,6 +305,59 @@ public boolean allowIvaratorFiltering() {
             return true;
         }
         
+        public String getWkt() {
+            String wkt = null;
+            
+            if (name.equals("within_bounding_box")) {
```
Comment (apmoriarty) [inline warehouse/query-core/src/main/java/datawave/query/jexl/functions/GeoFunctionsDescriptor.java]: Constant for `within_bounding_box` ?

File: warehouse/query-core/src/main/java/datawave/query/jexl/functions/GeoFunctionsDescriptor.java [hunk 2]
```diff
@@ -305,6 +305,59 @@ public boolean allowIvaratorFiltering() {
...
+                    // is the lower left longitude greater than the upper right longitude?
+                    // if so, we have crossed the anti-meridian and should split
+                    if (minLon > maxLon) {
+                        wkt = createGeometryCollection(createRectangle(minLon, 180.0, minLat, maxLat), createRectangle(-180.0, maxLon, minLat, maxLat))
+                                        .toText();
+                    } else {
+                        wkt = createRectangle(minLon, maxLon, minLat, maxLat).toText();
+                    }
+                }
+            } else if (name.equals("within_circle")) {
```
Comment (apmoriarty) [inline warehouse/query-core/src/main/java/datawave/query/jexl/functions/GeoFunctionsDescriptor.java]: Constant for `within_circle`?

File: warehouse/query-core/src/main/java/datawave/query/jexl/functions/GeoFunctionsDescriptor.java [hunk 3]
```diff
@@ -368,29 +421,9 @@ private boolean isAbstractGeometryType(String field, MetadataHelper helper) {
...
-                try {
-                    GeoPoint c = GeoPoint.decodeZRef(new GeoType().normalize(center));
-                    double radius = GeoNormalizer.parseDouble(args.get(2).image);
-                    
-                    wkt = createCircle(c.getLongitude(), c.getLatitude(), radius).toText();
-                } catch (IllegalArgumentException | OutOfRangeException | ParseException e) {
-                    log.warn("Encountered an error while parsing Geo function");
-                }
+            // only allow conversion to geowave function for the indexed geo types, not the individual lat/lon fields
+            if ((name.equals("within_bounding_box") && args.size() == 3) || name.equals("within_circle")) {
```
Comment (apmoriarty) [inline warehouse/query-core/src/main/java/datawave/query/jexl/functions/GeoFunctionsDescriptor.java]: Constants?

**Review by apmoriarty — DISMISSED** [review]

**Comment by jwomeara** [comment]
@ivakegg Uhh, what exactly got force pushed to this branch?

**Review by ivakegg — APPROVED** [review]

**Review by keith-ratcliffe — APPROVED** [review]
👍

**Pull request merged**